### DISCUSSION TITLE
sqlalchemy-firebird works for Interbase 7.5?

### DISCUSSION THREAD

**Discussion by thomaz-yuji** [pauldex/sqlalchemy-firebird#47] [Q&A]
Firstly, congratulations on this impressive project!

I'm wondering if it's feasible to utilize sqlalchemy-firebird with an Interbase 7.5 database, given its compatibility with Firebird 2.5 on Linux. Using pure fdb, i can easily connect using:

from fdb.fbcore import Connection, InternalError
from fdb.fbcore import ISOLATION_LEVEL_READ_COMMITED, isc_info_version, isc_info_page_size
 
 
```
class MyConnection(Connection):
 
    def __init__(self, db_handle, dpb=None, sql_dialect=3, charset=None,
                 isolation_level=ISOLATION_LEVEL_READ_COMMITED):
        try:
            super(MyConnection, self).__init__(db_handle, dpb, sql_dialect, charset,
                                                   isolation_level)
        except InternalError as Err:
            verstr = self.db_info(isc_info_version)
            x = verstr.split()
            if x[0].find('V') > 0:
                (x, self.__version) = x[0].split('V')
            elif x[0].find('T') > 0:
                (x, self.__version) = x[0].split('T')
            else:
                # Unknown version
                self.__version = '0.0.0.0'
            x = self.__version.split('.')
            self.__engine_version = float('%s.%s' % (x[0], x[1]))
            self.__page_size = self.db_info(isc_info_page_size)

con = fdb.connect(host='hostname', port=3050, database='path', user='sysdba', password='password', connection_class=MyConnection)

```

How can i use connection_class from fdb to use sqlalchemy-firebird?

**Comment by pauldex** [comment]
You **may** be able to connect using a firebird+fdb connection string.

If you're using Python 3.8+, you will need to install fdb to your virtual environment after installing sqlalchemy-firebird.  Using the same parameters as in your example above, your connection string would look like this:

db_uri = "firebird+fdb://sysdba:password@hostname:3050/path"
from sqlalchemy import create_engine
engine = create_engine(db_uri)

Then use the engine object as described in the SQLAlchemy documentation.

Please let us know if this is successful.
